Source/Diagnostics/WarpXOpenPMD.cpp

```diff
@@ -1219,46 +1302,95 @@ WarpXOpenPMDPlot::WriteOpenPMDFieldsAll ( //const std::string& filename,
     if ( (0 == lev) && first_write_to_iteration )
       SetupFields(meshes, full_geom);
 
-    amrex::Box const & global_box = full_geom.Domain();
-    auto const global_size = getReversedVec(global_box.size());
+        amrex::Box const & global_box = full_geom.Domain();
+        auto const global_size = getReversedVec(global_box.size());
+
+        int const ncomp = mf[lev].nComp();
+        for ( int icomp=0; icomp<ncomp; icomp++ ) {
+            std::string const & varname = varnames[icomp];
+
+            auto [varname_no_mode, mode_index] = GetFieldNameModeInt(varname);
+            bool var_in_theta_mode = mode_index != -1;
+            std::string field_name = varname_no_mode;
+            // std::string varname_no_mode = varname.substr(0,field_str_length);
+            std::string comp_name = openPMD::MeshRecordComponent::SCALAR;
+            // assume fields are scalar unless they match the following match of known vector fields
+            GetMeshCompNames( lev, varname_no_mode, field_name, comp_name );
+            if ( first_write_to_iteration )
+            {
+                if (comp_name == openPMD::MeshRecordComponent::SCALAR) {
+                    auto meshes_it = meshes.find(field_name);
+                    if (meshes_it == meshes.end()) {
+                        auto mesh = meshes[field_name];
+                        auto mesh_comp = mesh[comp_name];
+                        SetupMeshComp( mesh, full_geom, mesh_comp, var_in_theta_mode );
+                        detail::setOpenPMDUnit( mesh, field_name );
+                        auto relative_cell_pos = utils::getRelativeCellPosition(mf[lev]);     // AMReX Fortran index order
+#ifndef WARPX_DIM_RZ
+                        std::reverse( relative_cell_pos.begin(), relative_cell_pos.end() ); // now in C order
+#endif
+                        mesh_comp.setPosition( relative_cell_pos );
+                    }
+                } else {
+                    auto mesh = meshes[field_name];
```

Comment:
Can we remove some of the code duplication between `if` and `else` here?
Looks exactly the same now?

Reply:
Discussed: little helper function that takes:
- field_name
- comp_name
- meshes

and returns:
- mesh
- mesh_comp